core/chaincode/lifecycle/endorsment_info_test.go

```diff
@@ -243,4 +256,52 @@ var _ = Describe("ChaincodeEndorsementInfoSource", func() {
 			})
 		})
 	})
+
+	Context("when LifecycleV20 capability is not enabled", func() {
+		BeforeEach(func() {
+			fakeCapabilities.LifecycleV20Returns(false)
+			fakeLegacyImpl.ChaincodeEndorsementInfoReturns(&lifecycle.ChaincodeEndorsementInfo{
+				Version:           "legacy-version",
+				EndorsementPlugin: "legacy-plugin",
+				ChaincodeID:       "legacy-id",
+			}, fmt.Errorf("fake-error"))
+		})
+
+		It("wraps and returns the error", func() {
+			res, err := cei.ChaincodeEndorsementInfo("channel-id", "cc-name", fakeQueryExecutor)
+			Expect(err).To(MatchError("fake-error"))
+			Expect(res).To(Equal(&lifecycle.ChaincodeEndorsementInfo{
+				Version:           "legacy-version",
+				EndorsementPlugin: "legacy-plugin",
+				ChaincodeID:       "legacy-id",
+			}))
```

Comment:
Does it really return the error and the result?
Also check the It - "wraps and returns the error"

Reply:
Done